File: canvas_on_path.cpp
#include <bits/stdc++.h>

using namespace std;

int canvas_pos[100001], space_ls[100000], summ;
int main(){
    ios_base::sync_with_stdio(0), cin.tie(0);
    int n, m, k, num;
    cin >> n >> m >> k;
    for (int i = 0; i < m; i++) {
        cin >> num;
        canvas_pos[i] = num;
    }
    for (int j=1; j<m; j++){
        space_ls[j-1] = canvas_pos[j]-canvas_pos[j-1]-1;
    }
    
    sort(space_ls, space_ls+m-1);
    
    for(int i=0; i<max(m-k, 0); i++){
        summ+=space_ls[i];
    }
    
    cout<<summ;

    return 0;
}

File: canvas_on_path.py
n, m, k = input().split()
n, m, k = int(n), int(m), int(k)

space_ls = []
txt = input().split()
canvas_pos = [int(i) for i in txt]
for j in range(1, m):
    space_ls.append(canvas_pos[j]-canvas_pos[j-1]-1)

space_ls.sort()

for o in range(k-1): 
    if(len(space_ls)): space_ls.pop()
    else: break
summ = 0
for l in space_ls: summ+=l
print(summ)